test_scripts/API/ButtonSubscription/common_buttonSubscription.lua

```diff
@@ -0,0 +1,308 @@
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local test = require("user_modules/dummy_connecttest")
+local events = require('events')
+
+
+--[[ Variables ]]
+local m = actions
+m.hashId = {}
+m.wait = utils.wait
+m.cloneTable = utils.cloneTable
+
+m.buttons = {
+  "OK",
+  "PLAY_PAUSE",
+  "SEEKLEFT",
+  "SEEKRIGHT",
+  "TUNEUP",
+  "TUNEDOWN",
+  "PRESET_0",
+  "PRESET_1",
+  "PRESET_2",
+  "PRESET_3",
+  "PRESET_4",
+  "PRESET_5",
+  "PRESET_6",
+  "PRESET_7",
+  "PRESET_8",
+  "PRESET_9",
+  "SEARCH"
+}
+
+m.errorCode = {
+  "UNSUPPORTED_REQUEST",
+  "DISALLOWED",
+  "REJECTED",
+  "IN_USE",
+  "DATA_NOT_AVAILABLE",
+  "TIMED_OUT",
+  "INVALID_DATA",
+  "CHAR_LIMIT_EXCEEDED",
+  "INVALID_ID",
+  "DUPLICATE_NAME",
+  "APPLICATION_NOT_REGISTERED",
+  "OUT_OF_MEMORY",
+  "TOO_MANY_PENDING_REQUESTS",
+  "GENERIC_ERROR",
+  "USER_DISALLOWED",
+  "TRUNCATED_DATA",
+  "READ_ONLY"
+}
+
+--[[ Functions ]]
+function m.rpcSuccess(pAppId, pRpc, pButtonName)
+  local cid = m.getMobileSession(pAppId):SendRPC(pRpc, { buttonName = pButtonName })
+  EXPECT_HMICALL("Buttons." .. pRpc,{ appID = m.getHMIAppId(pAppId), buttonName = pButtonName })
+    :Do(function(_, data)
+        m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+      end)
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+  :Do(function(_, data)
+    m.hashId[pAppId] = data.payload.hashID
+  end)
+end
+
+function m.rpcUnsuccess(pAppId, pRpc, pButtonName, pResultCode)
+  local cid = m.getMobileSession(pAppId):SendRPC(pRpc, { buttonName = pButtonName })
+  EXPECT_HMICALL("Buttons." .. pRpc,{ appID = m.getHMIAppId(pAppId), buttonName = pButtonName })
+    :Times(0)
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = pResultCode })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+    :Times(0)
+end
+
+function m.buttonPress(pAppId, pButtonName)
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONDOWN", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonPress",
+    { name = pButtonName, mode = "SHORT", appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():SendNotification("Buttons.OnButtonEvent",
+    { name = pButtonName, mode = "BUTTONUP", appID = m.getHMIAppId(pAppId) })
+  m.getMobileSession(pAppId):ExpectNotification( "OnButtonEvent",
+    { buttonName = pButtonName, buttonEventMode = "BUTTONDOWN"},
+    { buttonName = pButtonName, buttonEventMode = "BUTTONUP"})
+    :Times(2)
+  m.getMobileSession(pAppId):ExpectNotification( "OnButtonPress",
+    { buttonName = pButtonName, buttonPressMode = "SHORT"})
+end
+
+function m.buttonPress_CUSTOM_BUTTON(pButtonName, pCustomButtonID, pAppID)
```

Comment:
There is no need to define buttonPressUnsuccess_CUSTOM_BUTTON, add parameter pCustomButtonID to buttonPressUnsuccess, in case pCustomButtonID will be empty the OnButtonEvent and OnButtonPress wiil be sent as without custom id